src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -88,6 +88,12 @@ public virtual string MetadataName
             }
         }
 
+        /// <summary>
+        /// Gets the token for this symbol as it appears in metadata. Most of the time this is 0,
+        /// as it is when the symbol is not loaded from metadata.
+        /// </summary>
+        public virtual int MetadataToken { get; }
```

Comment:
It seems like this adds a field to 'Symbol', which we likely don't want.

It seems like it would be good to make this abstract and to implement it to return 0 on all source symbols which directly inherit from `Symbol`.

Reply:
Sure thing, that shouldn't be a big... oh
![image](https://user-images.githubusercontent.com/754264/131425539-641fb7d8-043b-4857-b7f0-f316a6b1c0f0.png)

Reply:
Yeah, poked around at this a bit just now.. I think it would require a significant amount of rooting around to distinguish all the source symbols from the PE symbols and synthesized symbols and so on, so perhaps no need to make the base property 'abstract'.

It's a little unfortunate because if a new PE symbol kind were introduced, it would be easy to forget to override 'MetadataToken'. If you can think of anything that would make it easier to catch those kinds of issues, like an additional assert a PE symbol is loaded, for example, that might help.

Reply:
I had a look around and couldn't see anything obvious, but to say I don't know the codebase is an understatement :)